core/assignment_v2.go

```diff
@@ -0,0 +1,159 @@
+package core
+
+import (
+	"fmt"
+	"math"
+	"math/big"
+	"sort"
+)
+
+type BlobParameters struct {
+	CodingRate              uint
+	ReconstructionThreshold float64
+	NumChunks               uint
+}
+
+func (p BlobParameters) MaxNumOperators() uint {
+
+	return uint(math.Floor(float64(p.NumChunks) * (1 - 1/(p.ReconstructionThreshold*float64(p.CodingRate)))))
+
+}
+
+var (
+	ParametersMap = map[byte]BlobParameters{
+		0: {CodingRate: 8, ReconstructionThreshold: 0.22, NumChunks: 8192},
+	}
+)
+
+// AssignmentCoordinator is responsible for assigning chunks to operators in a way that satisfies the security
+// requirements of the protocol, as well as the constraints imposed by the specific blob version.
+type AssignmentCoordinatorV2 interface {
+
+	// GetAssignments calculates the full set of node assignments
+	GetAssignments(state *OperatorState, blobVersion byte, quorum QuorumID) (map[OperatorID]Assignment, error)
+
+	// GetAssignment calculates the assignment for a specific operator
+	GetAssignment(state *OperatorState, blobVersion byte, quorum QuorumID, id OperatorID) (Assignment, error)
+
+	// GetChunkLength determines the length of a chunk given the blob version and blob length
+	GetChunkLength(blobVersion byte, blobLength uint) (uint, error)
+}
+
+type StdAssignmentCoordinatorV2 struct {
+}
+
+var _ AssignmentCoordinatorV2 = (*StdAssignmentCoordinatorV2)(nil)
+
+func (c *StdAssignmentCoordinatorV2) GetAssignments(state *OperatorState, blobVersion byte, quorum QuorumID) (map[OperatorID]Assignment, error) {
```

Comment:
Why do we need this struct (as opposed to having methods that don't operate on a struct)? Is the idea that we may end up having multiple algorithms, and we can toggle between which algorithm we use by using a different struct that implements `AssignmentCoordinatorV2`?